homeassistant/components/simplisafe/binary_sensor.py

```diff
@@ -0,0 +1,279 @@
+"""Support for SimpliSafe binary sensors."""
+from simplipy.entity import EntityTypes
+from simplipy.websocket import EVENT_DOORBELL_DETECTED, EVENT_MOTION_DETECTED
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_MOISTURE,
+    DEVICE_CLASS_MOTION,
+    DEVICE_CLASS_OCCUPANCY,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.event import async_call_later
+
+from . import SimpliSafeEntity
+from .const import DATA_CLIENT, DOMAIN
+
+SUPPORTED_SENSOR_TYPES = [
+    EntityTypes.motion,
+    EntityTypes.entry,
+    EntityTypes.carbon_monoxide,
+    EntityTypes.smoke,
+    EntityTypes.leak,
+]
+
+HA_SENSOR_TYPES = {
+    EntityTypes.motion: DEVICE_CLASS_MOTION,
+    EntityTypes.entry: DEVICE_CLASS_DOOR,
+    EntityTypes.carbon_monoxide: DEVICE_CLASS_GAS,
+    EntityTypes.smoke: DEVICE_CLASS_SMOKE,
+    EntityTypes.leak: DEVICE_CLASS_MOISTURE,
+}
+
+SENSOR_MODELS = {
+    EntityTypes.motion: "Motion Sensor",
+    EntityTypes.entry: "Entry Sensor",
+    EntityTypes.carbon_monoxide: "Carbon Monoxide Detector",
+    EntityTypes.smoke: "Smoke Detector",
+    EntityTypes.leak: "Water Sensor",
+}
+
+MOTION_SENSOR_TRIGGER_CLEAR = 10
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up SimpliSafe binary sensors based on a config entry."""
+    simplisafe = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+
+    # Add sensor
+    async_add_entities(
+        [
+            SimpliSafeBinarySensor(simplisafe, system, sensor)
+            for system in simplisafe.systems.values()
+            for sensor in system.sensors.values()
+            if sensor.type in SUPPORTED_SENSOR_TYPES
+        ]
+    )
+
+    # Add low battery status entity for every sensor
+    async_add_entities(
+        [
+            SimpliSafeSensorBattery(simplisafe, system, sensor)
+            for system in simplisafe.systems.values()
+            for sensor in system.sensors.values()
+            if sensor.type in SUPPORTED_SENSOR_TYPES
+        ]
+    )
+
+    # Add motion sensor for cameras and doorbells
+    cameras = []
+    doorbells = []
+    for system in simplisafe.systems.values():
+        if (
+            "cameras" in system._location_info["system"]
```

Comment:
We should not be accessing a protected attribute here – this should be abstracted out in `simplisafe-python`.